demos/python_demos/colorization/colorization.py

```diff
@@ -0,0 +1,123 @@
+import cv2 as cv
+import numpy as np
+import os
+from argparse import ArgumentParser, SUPPRESS
+import xml.etree.ElementTree as ET
+import re
+import logging as log
+import glob
+import sys
+
+
+def build_arg():
+    parser = ArgumentParser(add_help=False)
+    in_args = parser.add_argument_group('Options')
+    in_args.add_argument('-h', '--help', action='help', default=SUPPRESS, help='Help with the script.')
+    in_args.add_argument("-m", "--model", help="Required. Path to .xml file with pre-trained model",
+                         required=True, type=str)
+    in_args.add_argument("-d", "--device",
+                         help="Optional. Specify target device for infer: CPU, GPU, FPGA, HDDL or MYRIAD. "
+                         "Default: CPU",
+                         default="CPU", type=str)
+    in_args.add_argument("-i", "--input", help="Optional. Path to a test video file.", type=str)
+    in_args.add_argument("-n", "--no_show", help="Optional. Disable display of results on screen.",
+                         action='store_true', default=False)
+    return parser
+
+
+def get_parameters(xml_path):
+    root = ET.parse(xml_path).getroot()
+    pattern = r'data\[([0-9]*\.*[0-9]*)\]'
+    scale_data = root.findall("./meta_data/cli_parameters/scale_values")[0]
+    mean_data = root.findall("./meta_data/cli_parameters/mean_values")[0]
+    scale_const = float(re.match(pattern, scale_data.attrib['value']).group(1))
+    mean_const = float(re.match(pattern, mean_data.attrib['value']).group(1))
+    return scale_const, mean_const
+
+
+def get_model_files(args_in):
+    config_xml_in = args_in.model
+    model_path = os.path.splitext(config_xml_in)[0]
+    weights_bin_in = model_path + ".bin"
+    level_up = os.path.split(os.path.split(model_path)[0])[0]
+    coeffs_in = level_up + "/" + os.path.split(level_up)[1] + ".npy"
+    return config_xml_in, weights_bin_in, coeffs_in
+
+
+if __name__ == '__main__':
+    log.basicConfig(format="[ %(levelname)s ] %(message)s", level=log.INFO, stream=sys.stdout)
+
+    args = build_arg().parse_args()
+    config_xml, weights_bin, coeffs = get_model_files(args)
+    scale, mean = get_parameters(config_xml)
+
+    log.info("Load network")
+    load_net = IENetwork(model=config_xml, weights=weights_bin)
+    exec_net = IECore().load_network(network=load_net, device_name=args.device)
+
+    input_blob = next(iter(load_net.inputs))
+    output_blob = next(iter(load_net.outputs))
+
+    n, c, h_in, w_in = load_net.inputs[input_blob].shape
+
+    cap = cv.VideoCapture(args.input if args.input else 0)
+    while cv.waitKey(1) < 0:
+        log.info("#############################")
+        hasFrame, original_frame = cap.read()
+        if not hasFrame:
+            cv.waitKey()
+            break
+        (h_orig, w_orig) = original_frame.shape[:2]
+
+        log.info("Preprocessing frame")
+        frame = cv.cvtColor(cv.cvtColor(original_frame, cv.COLOR_RGB2GRAY), cv.COLOR_GRAY2RGB)
```

Comment:
* The image here is `BGR`. It should be `frame = cv.cvtColor(cv.cvtColor(original_frame, cv.COLOR_BGR2GRAY), cv.COLOR_GRAY2RGB)`
* if cap uses cv.CAP_IMAGES and it gets gray image, `cvtColor()` will throw an error. I think it is essential to handle that because the demo is about colorization.

Reply:
@Wovchena  done